monix-execution/js/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -0,0 +1,104 @@
+package monix.execution.misc
+
+object Local {
+
+  type Context = scala.collection.immutable.Map[Key, Option[LocalContext]]
```

Comment:
`localContext.get` may return the value or null. When we do a `Local.get(key)` it gets flattened in that function so we won't get a `Some(None)`. I defined it like this because some LocalContext instance may register to the Local object state and may manage its existance through a Local class instance. If it wants to clear the LocalContext (which could be itself), it won't need to use the `clear()` function of the local nor set the value to null. I chose the Option to express that a LocalContext may still have its key still registered on the Local.Context Map which could be set or unset without having to unregister itself (by deleting its key from the Local.Context). Does that make sense?